### PR TITLE
feat: table compact styling

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

- Adds the compact styling to the table 
- Update docs to include table usage including this new feature

### To test:

https://app.clickup.com/t/1rr57f4

1. Add this PR to the [msq one](https://github.com/ArkEcosystem/marketsquare.io/pull/2456) and [explorer one](https://github.com/ArkEcosystem/explorer/pull/1011)
2. See if the table really applies the styling

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [x] I provided a screenshot of my changes to the component _(if applicable)_
-   [x] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [x] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: usage/ui.md [hunk 1]
```diff
@@ -110,6 +110,98 @@ import Modal from "@ui/js/modal";
...
+```
+
+1. Import the modal script in your `resources/js/app.js` file
+
+```js
+import Modal from "@ui/js/modal";
+
+window.Modal = Modal;
+```
+
```
Comment (leMaur) [inline usage/ui.md:204]: I think this part is related to the previous `### Modals` section 😅

**Review by leMaur — CHANGES_REQUESTED** [review]
@alfonsobries Tested on MSQ, all good to me! Just a little refactor on documentation, otherwise 👍🏼

File: usage/ui.md (hunk 1 above)
Comment (alfonsobries) [reply usage/ui.md:204]: 😅

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
@alfonsobries this affects the explorer styling, is that intended behaviour? the compact table becomes a lot larger than before

current:
image

after these changes:
image

**Comment by alfonsobries** [comment]
@ItsANameToo I should have created a PR for explorer from the beginning, sorry about that
 
I updated this PR to also work when the `compact-table` is added to the body to make it easy to integrate with explorer that has the toggle for compact and expanded. [PR](https://github.com/ArkEcosystem/explorer/pull/1011)

cc @leMaur if you have some time you can review explorer too

**Comment by leMaur** [comment]
@alfonsobries tested on Msq and it's OK but on Explorer the row height is 57px 🤔, is it wanted?
Screenshot 2021-11-30 at 18 14 39

**Comment by alfonsobries** [comment]
@leMaur 🤔  mm no it should have the same height that it always has, I am checking again

**Comment by alfonsobries** [comment]
@leMaur Did you recompile the assets? I just tested and the height is the one you mention before recompiling, once Explorer use the `_tables.css` from this PR the height is correct,

File: resources/assets/css/_tables.css [hunk 2]
```diff
@@ -760,6 +760,110 @@
...
+}
+
+@screen lg {
+    .table-container
+        table.table-compact.table-compact-until-lg
+        > tbody
+        > tr
+        > td.hoverable-cell
+        > div.table-cell-content,
+    body..table-compact.table-compact-until-lg
```
Comment (leMaur) [inline resources/assets/css/_tables.css:838]: @alfonsobries I think I found the culprit 😅. If I remove the dot and recompile, it works!
```suggestion
    body.table-compact.table-compact-until-lg
```

**Review by leMaur — APPROVED** [review]
👍🏼

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**